flat_mutation_reader.cc

```diff
@@ -955,6 +956,100 @@ make_flat_mutation_reader_from_fragments(schema_ptr schema, reader_permit permit
     return make_flat_mutation_reader_from_fragments(std::move(schema), std::move(permit), std::move(fragments), pr);
 }
 
+flat_mutation_reader
+make_filtering_reader(flat_mutation_reader rd, const dht::partition_range& pr, const query::partition_slice& slice) {
+    class reader : public flat_mutation_reader::impl {
+        flat_mutation_reader _rd;
+        const dht::partition_range* _pr;
+        const query::partition_slice* _slice;
+        dht::ring_position_comparator _cmp;
+        std::optional<clustering_ranges_walker> _ranges_walker;
+
+    public:
+        reader(flat_mutation_reader rd, const dht::partition_range& pr, const query::partition_slice& slice)
+            : flat_mutation_reader::impl(rd.schema(), rd.permit())
+            , _rd(std::move(rd))
+            , _pr(&pr)
```

Comment:
We should extend the existing filtering_reader rather than add a new one.

Reply:
This is not really a filtering reader, more like a slicing one. BTW @StarostaGit I don't see the the dropping of unselected columns, which is also part of slicing.

Reply:
@avikivity @denesb Will it be OK if I just give it a better name? E.g. `slicing_reader`?

Reply:
Yes, but note that this will collide with @tgrabiec 's flat mutation reader v2 series, which fixes up the existing pseudo slicing reader in `make_flat_mutation_reader_from_fragments()`. This latter can be replaced by this reader, can be done as a follow-up.

Reply:
OK, I can unify them afterwards. For now I renamed the one from this PR as `make_slicing_filtering_reader`